app/fruits/[fruitName]/actions.js

```diff
@@ -0,0 +1,26 @@
+'use server';
+
+import { cookies } from 'next/headers';
+import { getCookie } from '../../../util/cookies';
+import { parseJson } from '../../../util/json';
+
+export async function createComment(fruitId, comment) {
+  const fruitComments = parseJson(getCookie('fruitComments'));
+
+  const currentComments = Array.isArray(fruitComments)
+    ? fruitComments.filter((fruitComment) => fruitComment.id !== fruitId)
+    : [];
+
+  await cookies().set(
+    'fruitComments',
+    JSON.stringify(
+      [
+        ...currentComments,
+        {
+          id: fruitId,
+          comment,
+        },
+      ].sort((commentA, commentB) => commentA.id - commentB.id),
```

Comment:
instead of sorting, what about:

1. declare a new variable `const fruitToUpdate` using `.find()`
2. `if (fruitToUpdate)` -> mutate `fruitToUpdate`
3. `else` -> push object literal to `fruitComments`
4. `cookies().set()` with `fruitComments`

no spread operator + sort required, seems simpler

Reply:
This approach has some issues

1. we still need an extra variable fruitToUpdate
2. what if fruitComments is not an array? step 3 fails
3. what happens if fruitComments is undefined? step 4 also fails

 I just implemented it after fixing the problem mentioned, but I don't believe this is simpler. 

b2409c5

Reply:
> what if fruitComments is not an array? step 3 fails
> what happens if fruitComments is undefined? step 4 also fails

maybe these would be a good errors to leave in the action? (otherwise, if we handle them, it fails silently for students, and it's harder for them to debug)

Reply:
> 3. what happens if fruitComments is undefined? step 4 also fails

is this handled in the commit you linked above?

[`5f5e19e` (#15)](https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/pull/15/commits/5f5e19ebf8f50871e0209fc5f4d835a662d23153)

I guess it's still unhandled, and it's a good point that it's probably still needed

maybe after you [separate the `parseJson` into a new line](https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/pull/15/files/6d0acdb6af22201797bf9e53eda7f533b3997192#r1197012090), this will be easier to write simply

Reply:
handled with this pattern

```js
  const fruitCommentsCookie = getCookie('fruitComments');

  const fruitComments = !fruitCommentsCookie
    ? []
    : parseJson(fruitCommentsCookie);
```